src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -277,7 +277,8 @@ IndexerConnector::IndexerConnector(
     const std::function<void(
         const int, const std::string&, const std::string&, const int, const std::string&, const std::string&, va_list)>&
         logFunction,
-    const uint32_t& timeout)
+    const uint32_t& timeout,
+    const uint8_t workingThreads)
```

Comment:
I tested this and the problem is that the code will not work and it will be hard to detect the reason

---

We need some sort of mechanism to detect this on the tests:
- Use of asserts
  - Only works when compiled in debug
  - We need it to fail on the code we are using in production
- Use of exceptions
  - We need to be careful not to catch it and still fail to detect this
- Use of abort
  - We can add an if statement and an abort clause

Reply:
Thank you @sebasfalcone .

I've updated the code to overcome this situation.